File: Code/Project/XML_Parser.java
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class XML_Parser {

    public static void main(String[] args) {
        XML_Parser app = new XML_Parser();
        app.start();
    }

    private void start() {
        SparkSession spark = SparkSession
        .builder()
        .appName("Pubmed Article Set")
        .getOrCreate();

        DataFrame df = spark.read()
            .format("xml")
            .option("rowTag", "PubmedArticle")
            .load("truncatedPubmed.xml");

        df.show(5);
        df.printSchema();
    }
}

File: Code/Project/XML_Parser.scala
import org.apache.spark.sql.SparkSession
import com.databricks.spark.xml._


object XML_Parser {
  def main(args: Array[String]) {
    val spark = SparkSession.builder.appName("XML Parse").getOrCreate()
    val df = spark.read // sc.read   or   spark.read.format("com.databricks.spark.xml")
      .option("rowTag", "PubmedArticle")
      .xml("tmp/pubmed20n1000.xml") // .load("")
  }
}